src/base/static/components/place-detail/index.js

```diff
@@ -97,6 +99,38 @@ class PlaceDetail extends Component {
     this.props.model.attachmentCollection.add(attachment);
   }
 
+  // NOTE: Because we serialize our survey model collection before passing it
+  // down to the survey editor, we aren't able to (easily) pass down a
+  // reference to each survey model's save method. As a result, we have a
+  // special handler here to update survey models.
+  onSurveyModelSave(attrs, modelId) {
+    this.setState({ isSurveyEditFormSubmitting: true });
+    this.props.model.submissionSets[this.surveyType].get(modelId).save(attrs, {
+      success: () => {
+        this.setState({
+          isSurveyEditFormSubmitting: false,
+          surveyModels: serializeBackboneCollection(
+            this.props.model.submissionSets[this.surveyType],
+          ),
+        });
+      },
+    });
+  }
+
+  onSurveyModelRemove(modelId) {
+    if (confirm(messages.confirmRemove)) {
+      this.props.model.submissionSets[this.surveyType].get(modelId).destroy({
+        success: () => {
+          this.setState({
+            surveyModels: serializeBackboneCollection(
+              this.props.model.submissionSets[this.surveyType],
+            ),
+          });
+        },
+      });
+    }
+  }
+
```

Comment:
I'm not too sure about lines 102-133 here. This deviates from our pattern of passing down references to Backbone model methods to child components, but only because we'd have to add additional cruft to track a reference to each survey model's `.save()` method. It seemed easier to just create dedicated handlers for updating and deleting survey models.

Reply:
Is the "cruft" you speak of that we need to update our `state.surveyModels` after saving, based on the `this.props.model.submissionSets` collections that get updated after saving? If so, there should be a way to ensure that PlaceDetail's `state.surveyModels` get updated appropriately.

Ultimately, I see this being friction related to our use of Backbone models in our app state. I can understand if we want to punt this until we finish the port, then we can explore other options for state management like Redux. But I also want to balance that with having components that have clear boundaries and well defined abstractions.

Reply:
> Ultimately, I see this being friction related to our use of Backbone models in our app state

Right, I think some of this will be ironed out by using Redux.

The cruft would be due to the fact that each `PlaceDetailSurveyEditor` would need a reference to its Backbone model's `.save()` method (each survey response is represented by a separate model). When we serialize the survey collection, we're only serializing the attributes hash, and we lose all other aspects of the model.

Previously we didn't have a problem with this because we were either updating a single model (in the case of the `PlaceDetailEditor`) or we only needed access to a collection method (in the case of the `PlaceDetailSurvey`). The `PlaceDetailSurveyEditor` is different in the sense that it needs to know about the `save()` method of each individual model in the collection.

Reply:
Yeah, I think it's fine to punt this for now and address it when we introduce a better state management system.